src/helpers/solidity-extractor.js

```diff
@@ -0,0 +1,66 @@
+const fs = require('fs')
+const { promisify } = require('util')
+const readFile = promisify(fs.readFile)
+
+const modifiesStateAndIsPublic = de => {
+  const blacklist = ['internal', 'private', 'view', 'pure']
+
+  // space words to ensure they are not part of another word
+  return blacklist.filter(w => de.indexOf(` ${w} `) != -1).length == 0
```

Comment:
`w` needs to be clearer and we need to use strict equality to avoid JS weirdness (`===` / `!==`)